### ISSUE TITLE
Doesn't work with podman (docker-podman) because of write persmission

### ISSUE THREAD

**Issue by metal3d** [fyne-io/fyne-cross#41]
I'm using podman + docker-podman, it should be compatible.

Looking on the Dockerfile, it seems that you're using gosu to change the right + user id, but the directory continues to be owned by root:
```
fyne@ccd04cf246dd:/app/fyne-cross/tmp/windows-amd64$ ls -la .
total 8
drwxr-xr-x. 1 root root   46 Apr 21 19:45 .
drwxr-xr-x. 1 root root   26 Apr 21 19:45 ..
-rw-rw-r--. 1 root root 4886 Apr 21 19:45 fyne-completion.exe.ico
```

This way, the container wouldn't create files in the current directory.

I suggest the use of fixuid instead of using gosu: https://github.com/boxboat/fixuid

Then the command should not use `fyne_uid` but the standard `--user` option. That would be more efficient and would work with any container engine.

One note, you should suffix path volumes with "`:z` to avoid SELinux constraints :)

**Comment by metal3d** [comment]
I'm currently working on a patch to:
- use `fixuid` to avoid right problems
- using the `:z` suffix on `docker.go`
- remove the use of `fyne_uid`which is useless with `fixuid`

**Comment by metal3d** [comment]
At this time:
- no fixuid (i leave it without any touch, maybe later)
- for podman: I now find if podman is used instead of docker, so I set `--userns` option to the command line
- I did the change to append `:z` suffix to not have any problem with SELinux (it set label automatically)

see #42

**Comment by fiurthorn** [comment]
using `go install github.com/fyne-io/fyne-cross@develop` 
on results in

```
$ fyne-cross windows
[i] Target: windows/amd64
[i] Cleaning target directories...
[✓] "bin" dir cleaned: /home/fiurthorn/workspace/fyne/fyne-cross/bin/windows-amd64
[✓] "dist" dir cleaned: /home/fiurthorn/workspace/fyne/fyne-cross/dist/windows-amd64
[✓] "temp" dir cleaned: /home/fiurthorn/workspace/fyne/fyne-cross/tmp/windows-amd64
[i] Checking for go.mod: /home/fiurthorn/workspace/fyne/go.mod
[✓] go.mod found
chown: changing ownership of '/go': Operation not permitted
[✗] could not package the Fyne app: exit status 1
```

```
$ podman version
Version:      3.2.1
API Version:  3.2.1
Go Version:   go1.16.7
Built:        Thu Jan  1 01:00:00 1970
OS/Arch:      linux/amd64
```

```
$ cat /etc/lsb-release 
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=21.10
DISTRIB_CODENAME=impish
DISTRIB_DESCRIPTION="Ubuntu 21.10"
```

**Comment by Jacalz** [comment]
I see that you have moved your comment to this issue instead. I’ll just link to my response inside the PR thread: https://github.com/fyne-io/fyne-cross/pull/42#issuecomment-1046184041

**Comment by lucor** [comment]
As @Jacalz pointed out in the comment above the docker images were not published yet.
They are now available for testing, please see https://github.com/fyne-io/fyne-cross/pull/90 for the code and references.

**Comment by fiurthorn** [comment]
Thank you, I'll have a look next time.

**Comment by lucor** [comment]
Available on #90 for testing
